analytics/report_controller.py

```diff
@@ -0,0 +1,188 @@
+from analytics.models import (
+    Image, SearchEvent, SearchRatingEvent, ResultClickedEvent, DetailPageEvents,
+    AttributionReferrerEvent, DetailPageEvent,
+    UsageReport,SourceUsageReport, AttributionRefererReport,
+    TopSearchesReport, TopResultsReport
+)
+from sqlalchemy import func, distinct, Integer, and_
+from sqlalchemy.sql.expression import cast
+
+
+def generate_usage_report(session, start_time, end_time):
+    """ Get usage stats between start and end dates """
+    results_clicked = session.query(ResultClickedEvent).filter(
+        start_time < ResultClickedEvent.timestamp,
+        ResultClickedEvent.timestamp < end_time
+    ).count()
+    attribution_buttonclicks = session.query(DetailPageEvent).filter(
+        start_time < DetailPageEvent.timestamp,
+        DetailPageEvent.timestamp < end_time,
+        DetailPageEvent.event_type == DetailPageEvents.ATTRIBUTION_CLICKED
+    ).count()
+    survey_responses = session.query(DetailPageEvent).filter(
+        DetailPageEvent.timestamp > start_time,
+        DetailPageEvent.timestamp < end_time,
+        DetailPageEvent.event_type == DetailPageEvents.REUSE_SURVEY
+    ).count()
+    source_clicked = session.query(DetailPageEvent).filter(
+        DetailPageEvent.timestamp > start_time,
+        DetailPageEvent.timestamp < end_time,
+        DetailPageEvent.event_type == DetailPageEvents.SOURCE_CLICKED
+    ).count()
+    creator_clicked = session.query(DetailPageEvent).filter(
+        DetailPageEvent.timestamp > start_time,
+        DetailPageEvent.timestamp < end_time,
+        DetailPageEvent.event_type == DetailPageEvents.CREATOR_CLICKED
+    ).count()
+    shared_social = session.query(DetailPageEvent).filter(
+        DetailPageEvent.timestamp > start_time,
+        DetailPageEvent.timestamp < end_time,
+        DetailPageEvent.event_type == DetailPageEvents.SHARED_SOCIAL
+    ).count()
+    sessions = session.query(
+        func.count(
+            distinct(SearchEvent.session_uuid)
+        ).filter(
+            SearchEvent.timestamp > start_time,
+            SearchEvent.timestamp < end_time
+        )
```

Comment:
Just a reminder that `session_uid` is currently cached for 7 days for each user and I'll let you know when it's fixed.